### PR TITLE
Redesign edit profile page

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

**Comment by johngian** [comment]
Initial rebased branch combining both frontend with discrete commits is here:
https://github.com/johngian/mozillians/commits/profile-ui-latest-rebased

**Comment by johngian** [comment]
I squashed all the UI to one commit since there was a single bug for the frontend stuff. Kept all the commit messages to describe the functionality.

**Comment by johngian** [comment]
@akatsoulas Do you think that I should also squash 3c465235630b3fc8b31db1a415c1744aa35d92dd with 6cdbcf63ca3f42dce84f1a7be77e0846f433e1ba to have a single commit for the backend?

**Comment by johngian** [comment]
@akatsoulas @comzeradd Please take a look to see if something went wrong in the rebase because there where *many* conflicts.

**Comment by akatsoulas** [comment]
@johngian makes sense to me

File: mozillians/static/mozillians/js/profile_edit.js [hunk 1]
```diff
@@ -1,4 +1,65 @@
...
+    $('.settings-nav > li > a').on('click', function(e) {
+        e.preventDefault();
+        $('.settings-tab').hide();
+        $('.settings-nav > li').removeClass('active');
+        $(this).parent('li').addClass('active');
+        var content = $(this).attr('href');
+        $(content + '-tab').show();
+        location.hash = content;
+    });
+    if (screen.width < mobile_breakdown) {
```
Comment (akatsoulas) [inline mozillians/static/mozillians/js/profile_edit.js]: Continuing the discussion from the previous PR.
Even when I am using a direct url like `profile/edit/#mygroups` the settings tab is not visible.

**Comment by johngian** [comment]
PR still has some issues. Some of them can be tackled while QA works on updating the tests that gives us time to do some polishing. The rest of them are blocking this PR. FYI most of the blockers are already mentioned in previous reviews. 

Some things might also be broken because of the rebasing that's why we need to take a look.

* Blocking
```
* TODO Issue with privacy buttn in external accounts save
** TODO JS buttons doesn't initialize hidden privacy input
*** Steps to reproduce:
*** Go to external accounts
*** Add >2 external accounts
*** Inspect hidden privacy element
*** Default value = " "
* TODO Wrong position for privacy errors http://pix.toile-libre.org/upload/original/1441800668.png
* TODO Delete account button is clickable by default
** Steps to reproduce
*** Edit your profile, go to delete section
*** Delete button is clickable
*** Toggle the value
*** Now the button is not clickable
* TODO Location form needs some love (registration) - too much space http://pix.toile-libre.org/upload/original/1441795455.png
```

* Non blocking
```
* TODO Add messages on profile creation to inform user to fill the profile
* TODO Persona button still misaligned (compare with prod)
* TODO Front page vertical line/alignment changed (need to double check UI with more data)
* TODO Misaligned (view groups, functional areas) http://pix.toile-libre.org/upload/original/1441793788.png
* TODO Fix UI for link contribution story
* TODO Fix UI for country/region/city
* TODO Fix next per section not per tab
* TODO Contribution section has different types of textboxes http://pix.toile-libre.org/upload/original/1441794640.png
```

File: mozillians/templates/phonebook/includes/profile_edit_basic.html [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+          I acknowledge that deleting my account will make it tougher
+          for me to show off my contributions and easily be found by
+          other members of the Mozilla community.
+        {% endtrans %}
+      </fieldset>
+    </div>
+    <div class="panel-footer">
+      <a href="{{ url('phonebook:profile_confirm_delete') }}" id="delete-profile"
+         class="btn btn-danger delete">
+        {{ _('Delete Profile') }}
```
Comment (akatsoulas) [inline mozillians/templates/phonebook/includes/profile_edit_basic.html]: This button should be disabled by default.

**Comment by comzeradd** [comment]
I'm pushing small commits on branch [comzeradd:profile-ui-2](https://github.com/comzeradd/mozillians/commits/profile-ui-2) that fix each of the pending issues.

**Comment by comzeradd** [comment]
Updated branch [comzeradd:profile-ui-3](https://github.com/comzeradd/mozillians/commits/profile-ui-3)

**Comment by johngian** [comment]
@akatsoulas Added latest changes and updated notes above. Do you want take a look or should I merge ?

**Comment by johngian** [comment]
@akatsoulas bump

File: mozillians/phonebook/forms.py [hunk 3]
```diff
@@ -256,10 +267,43 @@ def clean(self):
...
+        required=False,
+        label=_lazy(u'When did you get involved with Mozilla?'),
+        widget=MonthYearWidget(years=range(1998, datetime.today().year + 1),
+                               required=False))
+
+    class Meta:
+        model = UserProfile
+        fields = ('title', 'privacy_title',
+                  'date_mozillian', 'privacy_date_mozillian',
+                  'story_link', 'privacy_story_link')
```
Comment (akatsoulas) [inline mozillians/phonebook/forms.py]: Please add a comma at the end of the tuple.

File: mozillians/phonebook/forms.py [hunk 4]
```diff
@@ -226,6 +211,32 @@ def clean_skills(self):
...
+    savecountry = forms.BooleanField(label=_lazy(u'Required'),
+                                     initial=True, required=False,
+                                     widget=forms.CheckboxInput(attrs={'disabled': 'disabled'}))
+    saveregion = forms.BooleanField(label=_lazy(u'Save'), required=False, show_hidden_initial=True)
+    savecity = forms.BooleanField(label=_lazy(u'Save'), required=False, show_hidden_initial=True)
+
+    class Meta:
+        model = UserProfile
+        fields = ('timezone', 'privacy_timezone', 'privacy_geo_city', 'privacy_geo_region',
+                  'privacy_geo_country')
```
Comment (akatsoulas) [inline mozillians/phonebook/forms.py]: Comma missing

File: mozillians/phonebook/forms.py (hunk 3 above)
Comment (akatsoulas) [inline mozillians/phonebook/forms.py]: This seems to have multiple occurrences.

**Comment by akatsoulas** [comment]
Haven't tested this. The changes in the back-end look good code-wise. Since you have reviewed the UI multiple times, I skipped it completely.  Having said that, I still have a pending comment on `profile_edit.js`. For me it seems to be working fine without those lines. 

r+wc :beers: :rocket:  really nice job.

**Comment by johngian** [comment]
I've tested this branch and it seems to be OK in terms of functionality.
@comzeradd Any comments about @tasos comment?

**Comment by comzeradd** [comment]
The specific line is needed for the mobile view to display properly in cases where a user opens a url with a location.hash (eg. #groups) that it's not visible on mobile. Otherwise user gets an empty page.

**Comment by akatsoulas** [comment]
After doing some testing it seems that the user gets the correct page (eg. #mygroups). The only problem is that it's collapsed. If you click the `Show all settings` it correctly expands the `mygroups` section. This happens without those lines. If we leave those lines there and you try to open the `mygroups` hash then you get an open page starting from the basic info.

**Comment by comzeradd** [comment]
Yes, that's the intended behavior. A user should never get an empty page with just a reveal button. This js code ensures that profile and location settings are always visible on mobile.

**Comment by akatsoulas** [comment]
I agree, the user should never get a page with an expand button. The solution is to have the page expanded to the correct hash and not collapsed. With the solution that you are proposing to this PR the user gets all the profile page from the top, which is the basic info, and not the correct hash (eg #mygroups).

**Comment by comzeradd** [comment]
Yes, the user should always get the mobile view of Settings regardless on how he landed there. Otherwise he'll struggle to realize how to get to Profile settings without the sidebar.

**Comment by comzeradd** [comment]
@johngian we reviewed the registration page with @ppapadeas and it just needs a minor fix for now. Can you cherry-pick the [last commit](https://github.com/comzeradd/mozillians/commit/6a9a732d8f0b76084668118397b4f2c86b1dec27) on [my branch](https://github.com/comzeradd/mozillians/commits/profile-ui-redesign).

**Comment by akatsoulas** [comment]
I am not quite sure we are talking about the same thing. Probably, I am not understanding something right. I am going to re-check this thoroughly. That being said, one last question, can't we use a media selector for this?

**Comment by comzeradd** [comment]
On mobile view we don't use `location.hash` to amend the view. So my take is that we should ignore it even when it's present. That's why unfortunately we can't use a media selector to do this. We can only do this in js.

**Pull request merged**